component/src/main/java/com/iluwatar/component/GameObject.java

```diff
@@ -0,0 +1,87 @@
+package com.iluwatar.component;
+
+import java.util.ArrayList;
+
+/**
+ * GameObject is a class for all object in the game.
+ * It was constructed by a collection of component.
+ */
+
+public class GameObject {
+  private int velocity;
+  private int positionOFx;
+  private int positionOFy;
+  ArrayList<Component> componentArrayList;
+
+  /**
+   * Constructor for GameObject
+   * @param componentArrayList is the list of this object contains
+   */
+
+  public GameObject(ArrayList<Component> componentArrayList){
+    this.componentArrayList=new ArrayList<>();
+    this.componentArrayList.addAll(componentArrayList);
+  }
+
+  /**
+   * setter for velocity
+   * @param velocity is the velocity of this object
+   */
+
+  public void setVelocity(int velocity) {
+    this.velocity = velocity;
+  }
+
+  /**
+   * getter for velocity
+   */
+
+  public int getVelocity() {
+    return velocity;
+  }
+
+  /**
+   * setter for PositionOFx
+   * @param positionOFx is the PositionOFx of this object
+   */
+
+  public void setPositionOFx(int positionOFx) {
```

Comment:
```suggestion
  public void setPositionOfX(int positionOfX) {
```

Reply:
To follow camel-casing properly, I think that ^ should be used. Check elsewhere too.